src/BizHawk.Emulation.Cores/Computers/Amiga/PUAE.cs

```diff
@@ -0,0 +1,261 @@
+﻿using BizHawk.Common;
+using BizHawk.Common.CollectionExtensions;
+using BizHawk.Common.StringExtensions;
+using BizHawk.Emulation.Common;
+using BizHawk.Emulation.Cores.Waterbox;
+
+using System;
+using System.Collections.Generic;
+
+namespace BizHawk.Emulation.Cores.Computers.Amiga
+{
+	[PortedCore(
+		name: CoreNames.PUAE,
+		author: "UAE Team",
+		portedVersion: "5.0.0",
+		portedUrl: "https://github.com/libretro/libretro-uae")]
+	public partial class PUAE : WaterboxCore
+	{
+		private LibPUAE _puae;
+		private static string _chipsetCompatible = "";
+
+		[CoreConstructor(VSystemID.Raw.Amiga)]
+		public PUAE(CoreLoadParameters<object, PUAESyncSettings> lp)
+			: base(lp.Comm, new Configuration
+			{
+				DefaultWidth          = LibPUAE.PAL_WIDTH,
+				DefaultHeight         = LibPUAE.PAL_HEIGHT,
+				MaxWidth              = LibPUAE.PAL_WIDTH,
+				MaxHeight             = LibPUAE.PAL_HEIGHT,
+				MaxSamples            = 2 * 1024,
+				SystemId              = VSystemID.Raw.Amiga,
+				DefaultFpsNumerator   = 50,
+				DefaultFpsDenominator = 1
+			})
+		{
+			_syncSettings = lp.SyncSettings ?? new();
+
+			_puae = PreInit<LibPUAE>(new WaterboxOptions
+			{
+				Filename                   = "puae.wbx",
+				SbrkHeapSizeKB             = 5 * 512,
+				SealedHeapSizeKB           = 10 * 1024,
+				InvisibleHeapSizeKB        = 10 * 1024,
+				PlainHeapSizeKB            = 10 * 1024,

```

Comment:
Although not strictly necessary, if possible we should trim the various heap sizes before release as it does improve performance a bit.

Reply:
These are temporary until I figure out the maximum I need for the most beefy machine config.